src/algebra/algebra/subalgebra.lean

```diff
@@ -298,6 +298,18 @@ theorem injective_cod_restrict (f : A →ₐ[R] B) (S : subalgebra R B) (hf : 
   function.injective (f.cod_restrict S hf) ↔ function.injective f :=
 ⟨λ H x y hxy, H $ subtype.eq hxy, λ H x y hxy, H (congr_arg subtype.val hxy : _)⟩
 
+/-- Restrict an injective algebra homomorphism to an algebra isomorphism -/
+noncomputable def alg_equiv_range (f : A →ₐ[R] B) (hf : function.injective f) :
+  A ≃ₐ[R] f.range :=
+alg_equiv.of_bijective (f.cod_restrict f.range (λ x, f.mem_range.mpr ⟨x, rfl⟩))
+⟨(f.injective_cod_restrict f.range (λ x, f.mem_range.mpr ⟨x, rfl⟩)).mpr hf,
+  λ x, Exists.cases_on (f.mem_range.mp (subtype.mem x)) (λ y hy, ⟨y, subtype.ext hy⟩)⟩
```

Comment:
Should this be called `alg_equiv.of_injective` to match [`linear_equiv.of_injective`](https://leanprover-community.github.io/mathlib_docs/linear_algebra/basic.html#linear_equiv.of_injective)?

Reply:
You should also add a lemma that says `alg_equiv.of_injective f hf x = f x`, possibly marked `simp`